apps/admin/src/app/(api)/api/v1/csv/route.ts

```diff
@@ -0,0 +1,68 @@
+import { getElectricitySensorIdForUser, getEnergyForSensorInRange } from "@energyleaf/db/query";
+import * as csv from "csv/sync";
+import { type NextRequest, NextResponse } from "next/server";
+
+interface Props {
+    userId: string;
+    userHash: string;
+    start?: string;
+    end?: string;
+}
+
+export async function POST(req: NextRequest) {
+    const { userId, userHash, start, end } = (await req.json()) as Props;
+
+    if (!userId || !userHash) {
+        return NextResponse.json(
+            {
+                error: "Invalid user",
+            },
+            {
+                status: 401,
+            },
+        );
+    }
+
+    const base64UserId = Buffer.from(userId).toString("base64url").slice(0, 6);
```

Comment:
Wozu die User-ID? Ist das die Authentifizierung?

Reply:
Die UserId wird ja an den Server mit übergeben, da es vom Client aufgerufen werden kann haben wir einfach noch einen Hash um etwas Bots abzuhalten, so kann man allerdings auch eine eigene Cli bauen die im regelmäßigen Abstand die Daten runterlädt.

Denke das ist ein guter Kompromiss

Reply:
Wenn wir den eigenen Download nicht zulassen wollen, können wir auch einen "short lived" jwt generieren und diesen Verifizieren.

Da bin ich ganz offen.

Reply:
D. h., wenn ich die User-ID kenne, kann ich mir einfach so die Daten herunterladen? Klingt jetzt nicht so super sicher, gerade weil wir noch einige (ältere) IDs haben, die noch einfache Zahlen sind.
Macht es vllt Sinn, das noch ein wenig abzusichern? Man könnte ja z. B. ein Secret definieren, den man beim Erzeugen des Hashs benutzen. Vllt gibt es ja auch noch weitere Verfahren, die man gut anwenden könnte. Bin leider nicht so in Kryptographie drin

Reply:
Okay ich passe das mal an.